V3 dcl mode implementation (#1366)
luxonis/depthai-core: examples/python/DynamicCalibration/ReadMe.md

> # 🔧 DepthAI Dynamic Calibration Interactive script

Reviewer: inconsistent file naming convention: `ReadMe` instead `Readme` or `README`

An interactive calibration and diagnostics tool for OAK cameras using [DepthAI](https://github.com/luxonis/depthai-core). This script provides a real-time visual interface to:

- Trigger dynamic stereo calibration
- View calibration quality metrics (rotation error, depth improvements)
- Visualize depth disparity
- Overlay health bar indicators and coverage information

> - Flash and apply calibrations directly on the device

Reviewer: ```suggestion
- Flash and apply calibrations directly to the device
```

---

## 📸 Features

> - ✅ Real-time disparity visualization (`cv2.COLORMAP_JET`)

Reviewer: `cv2.COLORMAP_JET` shouldn't be mentioned as feature IMO. It's implementation detail what colormap we are using.. It could also be any other.

I also hope that we are not using `cv2.COLORMAP_JET` as is, but with disparity value 0 mapped to 0,0,0 instead blue?

- ✅ Calibration result summaries printed to screen
- ✅ Health bar overlays for depth accuracy
- ✅ Live keyboard control (see key commands)
- ✅ Coverage map overlay per frame
- ✅ Full DepthAI pipeline using left/right mono cams, stereo, and dynamic calibration

---

## 💻 Key Commands

> These can be pressed anytime in the GUI window:

Reviewer: `These` what exactly?

> These can be pressed anytime in the GUI window:

Reviewer: ```suggestion
The following keys can be pressed at any time within the OpenCV GUI window (also shown live on the left camera frame overlay)```

```
[c] → Calibration quality check  
[r] → Recalibrate  
[a] → Force calibration check  
[d] → Force recalibrate  
[n] → Apply new calibration  
[o] → Apply old calibration  
[l] → Flash new calibration  
[k] → Flash old calibration  
[q] → Quit  
```

> These are also shown live on the left camera frame overlay.

Reviewer: Remove if above suggestion is accepted

---

## 🚀 Setup & Usage

### 1. Install Dependencies

> You need:

Reviewer: Remove, it's clear that requirements are needed.

- Python 3.7+
- DepthAI > 3.0
- OpenCV
- NumPy

```bash

> pip install depthai opencv-python numpy

Reviewer: ```suggestion
python3 -m pip install depthai opencv-python numpy
```

```

### 2. Run the Code

```bash

> python dynamic_calibration_interactive.py

Reviewer: ```suggestion
python3 dynamic_calibration_interactive.py
```

```

---

## 🧠 Internals

- The viewer listens to the `DynamicCalibration` node for calibration results.
- Results are parsed, filtered, and visualized using OpenCV.
- A health bar shows how calibration changes affect depth error at 1m, 2m, 5m, and 10m.

> - Calibrations can be flashed to EEPROM or applied temporarily.

Reviewer: ```suggestion
- Updated calibration can be permanently stored on device or applied only temporarily until the session finishes.
```
